common/common.go

```diff
@@ -83,14 +83,13 @@ func LoadBuildConfigFromProvenance(provenance *slsa.Provenance) (*BuildConfig, e
 		return nil, fmt.Errorf("the provenance must have exactly one Subject, got %d", len(provenance.Subject))
 	}
 
-	proParams := provenance.Predicate.Invocation.Parameters
 	config := BuildConfig{
-		Repo:                     proParams.Repository,
-		CommitHash:               proParams.CommitHash,
-		BuilderImage:             proParams.BuilderImage,
-		Command:                  proParams.Command,
-		OutputPath:               provenance.Subject[0].Name,
-		ExpectedBinarySha256Hash: provenance.Subject[0].Digest.Sha256,
+		Repo:                     provenance.Predicate.Materials[1].URI,
```

Comment:
Imo no. Provenance files must be follow the amber buildType schema, which requires this. They are validated against the amber buildType before being parsed into the `Provenance` struct. Hence at this point in the code, this shape is expected. 

Of course there is the possibility of a bug leading to this somehow not being the case. But imo that case should be (and is) guarded against in tests.

Reply:
This is a public function and there is no guarantee that it will always be called after parsing and validating a provenance file. In general, each function has to check its input for validity assumptions first.

Reply:
In that case, shall we move the functionality implemented by this block of code into the SLSA module's `ParseProvenanceFile` function? It is the only function that ever calls this one, and for the foreseeable future will be. It already checks it's inputs. — This would prevent us from needing to maintain & validate a public `Provenance` struct that mirrors the public `BuildConfig` struct.

For context, that approach was my initial implementation, I switched to this following your request to closely model the existing code's structure. It is also worth noting that the implementation of the `LoadBuildConfigFromProvenance` function that precedes this PR also does not validate the number of subjects and materials, or whether keys are empty.

Reply:
The plan is to add some wrappers to the repo, for using authorization logic. The wrappers will need to [use some of the functionality in this repo](https://github.com/google-research/raksha/blob/6da01e3d3935a136f062837313d36dc28e20406a/rust/tools/authorization-logic/transparent_release_prototype/transparent_release.auth_logic#L167). I think a `LoadBuildConfigFromProvenance` that takes a file path as input should work for that use case too.

Reply:
> Are you suggesting to change ParseProvenanceFile to return BuildConfig? In that case, BuildConfig has to be moved to slsa.go too, right? To avoid a circular dependency.

With this change, slsa.go would import BuildConfig from common.go, yes. However common.go would no longer need to import slsa.go. Hence no circular dependency. 

> Alternatively, LoadBuildConfigFromProvenance could take a file path as input, and call ParseProvenanceFile before building an instance of BuildConfig. That allows keeping the internal representation BuildConfig separate from, slsa.go and keep that as a more coherent package. Can we still dedup the test with this approach? WDYT?

I'm also okay with this approach. Worth noting however that `ParseProvenanceFile` is all that slsa.go does. Hence we'd be essentially moving it into common.go completetly. Imo not a bad thing however. Imo this is the best way.

Reply:
@rbehjati I added validation for each LoadBuildConfigFromProvenance for now. Should we proceed with this, and look at refactoring for simplifying tests later? 

Imo that bit goes beyond the scope of this PR, which is just swapping out the existing logic.

Reply:
Yes. Let's do the refactoring in a separate PR :)